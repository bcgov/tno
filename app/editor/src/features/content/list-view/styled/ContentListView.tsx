import { FormPage } from 'components/form';
import styled from 'styled-components';

export const ContentListView = styled(FormPage)`
  min-height: fit-content;

  hr {
    width: 100%;
    height: 0.75em;
    border: none;
    background-color: #003366;
  }

  .content-list {
    border-radius: 4px;
    box-shadow: 0px 1px 2px rgba(0, 0, 0, 0.2);
    table {
      background: transparent;
    }

    div[role='rowgroup'] {
      min-height: 100px;
      max-height: calc(100vh - 600px);
      overflow-y: scroll;
      overflow-x: hidden;
    }
  }

  .content-actions {
    button {
      display: block;
    }
  }

  .top-pane {
    display: flex;
    flex-direction: column;
    flex: 1 1 0%;
  }

  .bottom-pane {
    display: flex;
    flex-direction: column;
    flex: 1 1 0%;
  }

  .h-status {
    text-align: center;
    align-items: center;
    justify-content: center;
  }

  .h-publishedOn {
    text-align: center;
    align-items: center;
    justify-content: center;
  }

  .h-use {
    text-align: center;
    align-items: center;
    justify-content: center;
  }
`;
