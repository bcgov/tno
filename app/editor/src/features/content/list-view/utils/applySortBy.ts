import { ISortBy } from '../interfaces';

/**
 * Creates an array of sort parameters from the provided sorting information.
 * Cleans up the data to ensure it matches what is expected by the API.
 * @param sortBy An array of sort objects.
 * @returns An array of sort parameters.
 */
export const applySortBy = (sortBy?: ISortBy[]) => {
  if (sortBy === undefined || sortBy.length === 0) return undefined;

  var sort: string[] = [];
  for (let i = 0; i < sortBy.length; i++) {
    let column = sortBy[i].id;
    if (column === 'section') {
      sort.push(`PrintContent.${column}${sortBy[i].desc ? ' desc' : ''}`);
      sort.push(`page${sortBy[i].desc ? ' desc' : ''}`);
    } else {
      sort.push(`${column}${sortBy[i].desc ? ' desc' : ''}`);
    }
  }
  return sort;
};
