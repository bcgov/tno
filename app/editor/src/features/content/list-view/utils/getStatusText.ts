import { ContentStatusName } from 'hooks';

/**
 * Provides a different status text for some values.
 * @param status The current status.
 * @returns A status text.
 */
export const getStatusText = (status: ContentStatusName) => {
  switch (status) {
    case ContentStatusName.Unpublish:
      return 'Unpublishing';
    case ContentStatusName.Publish:
      return 'Indexing';
    default:
      return status;
  }
};
