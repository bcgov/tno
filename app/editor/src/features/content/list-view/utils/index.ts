export * from './makeFilter';
export * from './setTimeFrame';
