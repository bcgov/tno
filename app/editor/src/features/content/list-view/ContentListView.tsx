import { FormPage } from 'components/form/formpage';
import { useCombinedView, useTooltips } from 'hooks';
import { IContentModel } from 'hooks/api-editor';
import React from 'react';
import { useNavigate, useParams } from 'react-router-dom';
import { SortingRule } from 'react-table';
import { useApp, useContent } from 'store/hooks';
import { Button, ButtonVariant, Col, Page, PagedTable, Row, Show } from 'tno-core';

import { FormPicker } from '../form';
import { ContentFilter } from '.';
import { columns, defaultPage } from './constants';
import { IContentListFilter } from './interfaces';
import * as styled from './styled';
import { makeFilter } from './utils';

export const ContentListView: React.FC = () => {
  const [{ userInfo }, { isUserReady }] = useApp();
  const { id } = useParams();
  const [{ filter, filterAdvanced, content }, { findContent, storeFilter }] = useContent();
  const navigate = useNavigate();
  const combined = useCombinedView();
  useTooltips();

  const [loading, setLoading] = React.useState(false);
  const [activeId, setActiveId] = React.useState<number>(parseInt(id ?? '0'));

  // Set the page for the grid table.
  const page = !!content
    ? new Page(content.page - 1, content.quantity, content?.items, content.total)
    : defaultPage;
  const userId = userInfo?.id ?? '';

  React.useEffect(() => {
    setActiveId(parseInt(id ?? '0'));
  }, [id]);

  React.useEffect(() => {
    if (userId !== 0 && filter.userId === '' && filter.userId !== userId) {
      storeFilter({ ...filter, userId });
    }
  }, [userId, filter, storeFilter]);

  const fetch = React.useCallback(
    async (filter: IContentListFilter) => {
      try {
        setLoading(true);
        const data = await findContent(
          makeFilter({
            ...filter,
            ...filterAdvanced,
          }),
        );
        const page = new Page(data.page - 1, data.quantity, data?.items, data.total);

        // setPage(page);
        return page;
      } catch (error) {
        // TODO: Handle error
        throw error;
      } finally {
        setLoading(false);
      }
    },
    [filterAdvanced, findContent],
  );

  React.useEffect(() => {
    // Only make a request if the user has been set.
    if (isUserReady() && filter.userId !== '') {
      fetch(filter);
    }
    // Only want to make a request when filter or sort change.
    // 'fetch' regrettably changes any time the advanced filter changes.
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [filter, isUserReady()]);

  const handleChangePage = React.useCallback(
    (pi: number, ps?: number) => {
      if (filter.pageIndex !== pi || filter.pageSize !== ps)
        storeFilter({ ...filter, pageIndex: pi, pageSize: ps ?? filter.pageSize });
    },
    [filter, storeFilter],
  );

  const handleChangeSort = React.useCallback(
    (sortBy: SortingRule<IContentModel>[]) => {
      const sorts = sortBy.map((sb) => ({ id: sb.id, desc: sb.desc }));
      const same = sorts.every(
        (val, i) => val.id === filter.sort[i]?.id && val.desc === filter.sort[i]?.desc,
      );
      if (!same) {
        storeFilter({ ...filter, sort: sorts });
      }
    },
    [storeFilter, filter],
  );

  const handleRowClick = (content: IContentModel) => {
    setActiveId(content.id);
    navigate(`/contents/combined/${content.id}`);
  };

  return (
    <styled.ContentListView>
      <FormPage>
        <Row wrap="nowrap">
          <Col className="left-pane">
            <ContentFilter search={fetch} />
            <Row className="content-list">
              <PagedTable
                columns={columns}
                page={page}
                isLoading={loading}
                sorting={{ sortBy: filter.sort }}
                onRowClick={(row) => handleRowClick(row.original)}
                activeId={activeId}
                onChangePage={handleChangePage}
                onChangeSort={handleChangeSort}
              />
            </Row>
            <Row className="content-actions">
              <Button
                name="create"
                onClick={() => navigate('/snippets/0')}
                variant={ButtonVariant.secondary}
              >
                Create Snippet
              </Button>
              <Button
                name="create"
                onClick={() => navigate('/papers/0')}
                variant={ButtonVariant.secondary}
              >
                Create Print Content
              </Button>
              <div>Send to</div>
              <Button
                name="create"
                variant={ButtonVariant.secondary}
                disabled
                tooltip="Under Construction"
              >
                Front Pages
              </Button>
              <Button
                name="create"
                variant={ButtonVariant.secondary}
                disabled
                tooltip="Under Construction"
              >
                Top Stories
              </Button>
              <Button
                name="create"
                variant={ButtonVariant.secondary}
                disabled
                tooltip="Under Construction"
              >
                Commentary
              </Button>
            </Row>
          </Col>
          <Show visible={combined}>
            <Col className="right-pane">
              <FormPicker />
            </Col>
          </Show>
        </Row>
      </FormPage>
    </styled.ContentListView>
  );
};
