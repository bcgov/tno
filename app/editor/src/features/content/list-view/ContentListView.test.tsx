import { render, waitFor } from '@testing-library/react';
import { ContentStatusName, ContentTypeName } from 'hooks';
import { mockContent, TestWrapper } from 'test/utils';

import { ContentListView } from './ContentListView';

jest.mock('store', () => ({
  useAppDispatch: jest.fn(),
  useAppSelector: jest.fn(),
}));

const mockUseLookups = [
  {
    products: [{ name: 'Product 1', id: 1 }],
    licenses: [{ name: 'License 1', id: 1 }],
    ingestTypes: [{ name: 'Ingest 1', id: 1 }],
    users: [{ name: 'user', id: 1, displayName: 'displayName', username: 'test' }],
  },
];

const mockUseLookupOptions = [
  {
    products: [{ name: 'Product 1', id: 1 }],
    licenses: [{ name: 'License 1', id: 1 }],
    ingestTypes: [{ name: 'Ingest 1', id: 1 }],
    users: [{ name: 'user', id: 1, displayName: 'displayName', username: 'test' }],
    ingestTypeOptions: [],
    productOptions: [],
    sourceOptions: [],
    userOptions: [],
  },
];

const mockUseContent = [
  {
    filter: {},
    filterAdvanced: {},
    content: {
      items: [
        {
          headline: 'test headline',
          owner: { displayName: 'user@idir', username: 'test@idir' },
          otherSource: 'TEST',
          createdOn: '2022-05-12T16:11:15.756251Z',
          contentType: ContentTypeName.Snippet,
          status: ContentStatusName.Publish,
          product: {
            name: 'News',
          },
          page: 'A32',
        },
      ],
      page: 1,
      total: 1,
      quantity: 10,
    },
    page: 1,
    total: 1,
    quantity: 10,
  },
  { findContent: () => Promise.resolve(mockContent), storeFilter: {} },
];

const mockUseApp = [
  {
    userInfo: '',
    requests: [],
  },
  {},
];

jest.mock('hooks', () => ({
  useLookupOptions: () => mockUseLookupOptions,
}));

jest.mock('store/hooks', () => ({
  useApp: () => mockUseApp,
  useContent: () => mockUseContent,
  useLookup: () => mockUseLookups,
}));

it('renders correctly...', async () => {
  const { container } = render(
    <TestWrapper>
      <ContentListView />
    </TestWrapper>,
  );
  // TODO: This snapshot isn't waiting for the `findContent` to resolve.
  await waitFor(() => expect(container).toMatchSnapshot());
});

it('displays username without idir tag', async () => {
  const { getByText, queryByText } = render(
    <TestWrapper>
      <ContentListView />
    </TestWrapper>,
  );
  await waitFor(() => {
    expect(getByText('user')).toBeInTheDocument();
    expect(queryByText('@idir')).toBeFalsy();
  });
});

it('displays the date in the correct format', async () => {
  const { getByText } = render(
    <TestWrapper>
      <ContentListView />
    </TestWrapper>,
  );
  await waitFor(() => expect(getByText('05/12/2022')).toBeInTheDocument());
});

it('displays the Source column', async () => {
  const { getByText } = render(
    <TestWrapper>
      <ContentListView />
    </TestWrapper>,
  );
  await waitFor(() => expect(getByText('Source')).toBeInTheDocument());
});

it('displays the Designation column', async () => {
  const { getByText } = render(
    <TestWrapper>
      <ContentListView />
    </TestWrapper>,
  );
  await waitFor(() => expect(getByText('Designation')).toBeInTheDocument());
});

it('displays the status column', async () => {
  const { getByText } = render(
    <TestWrapper>
      <ContentListView />
    </TestWrapper>,
  );
  await waitFor(() => expect(getByText('Pub Date')).toBeInTheDocument());
});

it('displays the Use column', async () => {
  const { getByText } = render(
    <TestWrapper>
      <ContentListView />
    </TestWrapper>,
  );
  await waitFor(() => expect(getByText('Use')).toBeInTheDocument());
});
