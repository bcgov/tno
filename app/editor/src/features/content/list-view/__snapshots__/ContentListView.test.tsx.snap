// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly... 1`] = `
<div>
  <div
    class="sc-efBctP lfePom"
  >
    <div
      class="sc-bczRLJ kIuZMu"
    >
      <div
        class="sc-iAvgwm klZmki content-filter"
      >
        <div>
          <div
            class="sc-iBkjds bEngFb sc-himrzO VoNvn frm-in"
            direction="column"
          >
            <label
              class=""
              for="sel-mediaType"
            >
              Media Type
            </label>
            <div
              class="sc-ftvSup klGvED"
              direction="row"
            >
              <div
                class="sc-gXmSlM hjMLBz css-b62m3t-container"
                id="sel-mediaType"
              >
                <span
                  class="css-1f43avz-a11yText-A11yText"
                  id="react-select-2-live-region"
                />
                <span
                  aria-atomic="false"
                  aria-live="polite"
                  aria-relevant="additions text"
                  class="css-1f43avz-a11yText-A11yText"
                />
                <div
                  class="rs__control css-1s2u09g-control"
                >
                  <div
                    class="rs__value-container css-319lph-ValueContainer"
                  >
                    <div
                      class="rs__placeholder css-14el2xx-placeholder"
                      id="react-select-2-placeholder"
                    >
                      Select...
                    </div>
                    <div
                      class="rs__input-container css-6j8wv5-Input"
                      data-value=""
                    >
                      <input
                        aria-autocomplete="list"
                        aria-describedby="react-select-2-placeholder"
                        aria-expanded="false"
                        aria-haspopup="true"
                        autocapitalize="none"
                        autocomplete="off"
                        autocorrect="off"
                        class="rs__input"
                        id="react-select-2-input"
                        role="combobox"
                        spellcheck="false"
                        style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                        tabindex="0"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                  <div
                    class="rs__indicators css-1hb7zxy-IndicatorsContainer"
                  >
                    <span
                      class="rs__indicator-separator css-1okebmr-indicatorSeparator"
                    />
                    <div
                      aria-hidden="true"
                      class="rs__indicator rs__dropdown-indicator css-tlfecz-indicatorContainer"
                    >
                      <svg
                        aria-hidden="true"
                        class="css-tj5bde-Svg"
                        focusable="false"
                        height="20"
                        viewBox="0 0 20 20"
                        width="20"
                      >
                        <path
                          d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
                <input
                  name="mediaType"
                  type="hidden"
                  value=""
                />
              </div>
            </div>
            <input
              autocomplete="off"
              style="opacity: 0; width: 100%; height: 0px; position: absolute;"
              tabindex="-1"
              value=""
            />
          </div>
          <div
            class="sc-iBkjds bEngFb sc-himrzO VoNvn frm-in"
            direction="column"
          >
            <label
              class=""
              for="sel-user"
            >
              User
            </label>
            <div
              class="sc-ftvSup klGvED"
              direction="row"
            >
              <div
                class="sc-gXmSlM hjMLBz css-b62m3t-container"
                id="sel-user"
              >
                <span
                  class="css-1f43avz-a11yText-A11yText"
                  id="react-select-3-live-region"
                />
                <span
                  aria-atomic="false"
                  aria-live="polite"
                  aria-relevant="additions text"
                  class="css-1f43avz-a11yText-A11yText"
                />
                <div
                  class="rs__control css-1s2u09g-control"
                >
                  <div
                    class="rs__value-container css-319lph-ValueContainer"
                  >
                    <div
                      class="rs__placeholder css-14el2xx-placeholder"
                      id="react-select-3-placeholder"
                    >
                      Select...
                    </div>
                    <div
                      class="rs__input-container css-6j8wv5-Input"
                      data-value=""
                    >
                      <input
                        aria-autocomplete="list"
                        aria-describedby="react-select-3-placeholder"
                        aria-expanded="false"
                        aria-haspopup="true"
                        autocapitalize="none"
                        autocomplete="off"
                        autocorrect="off"
                        class="rs__input"
                        id="react-select-3-input"
                        role="combobox"
                        spellcheck="false"
                        style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                        tabindex="0"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                  <div
                    class="rs__indicators css-1hb7zxy-IndicatorsContainer"
                  >
                    <span
                      class="rs__indicator-separator css-1okebmr-indicatorSeparator"
                    />
                    <div
                      aria-hidden="true"
                      class="rs__indicator rs__dropdown-indicator css-tlfecz-indicatorContainer"
                    >
                      <svg
                        aria-hidden="true"
                        class="css-tj5bde-Svg"
                        focusable="false"
                        height="20"
                        viewBox="0 0 20 20"
                        width="20"
                      >
                        <path
                          d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
                <input
                  name="user"
                  type="hidden"
                  value=""
                />
              </div>
            </div>
            <input
              autocomplete="off"
              style="opacity: 0; width: 100%; height: 0px; position: absolute;"
              tabindex="-1"
              value=""
            />
          </div>
          <div
            class="sc-ikZpkk idFYWB frm-in rag"
            direction="col-row"
            name="timeFrame"
          >
            <div>
              <label
                class=""
                data-for="main-tooltip"
                data-tip="Date created"
              >
                Time Frame
              </label>
              <span
                class="sc-jIZahH dViiyh"
              >
                <input
                  class="sc-kLLXSd lgwhuj rad "
                  data-for="main-tooltip"
                  id="rad-timeFrame-0"
                  name="timeFrame"
                  type="radio"
                  value="0"
                />
                <label
                  for="rad-timeFrame-0"
                >
                  Today
                </label>
              </span>
              <span
                class="sc-jIZahH dViiyh"
              >
                <input
                  class="sc-kLLXSd lgwhuj rad "
                  data-for="main-tooltip"
                  id="rad-timeFrame-1"
                  name="timeFrame"
                  type="radio"
                  value="1"
                />
                <label
                  for="rad-timeFrame-1"
                >
                  24hrs
                </label>
              </span>
              <span
                class="sc-jIZahH dViiyh"
              >
                <input
                  class="sc-kLLXSd lgwhuj rad "
                  data-for="main-tooltip"
                  id="rad-timeFrame-2"
                  name="timeFrame"
                  type="radio"
                  value="2"
                />
                <label
                  for="rad-timeFrame-2"
                >
                  48hrs
                </label>
              </span>
              <span
                class="sc-jIZahH dViiyh"
              >
                <input
                  class="sc-kLLXSd lgwhuj rad "
                  data-for="main-tooltip"
                  id="rad-timeFrame-3"
                  name="timeFrame"
                  type="radio"
                  value="3"
                />
                <label
                  for="rad-timeFrame-3"
                >
                  All
                </label>
              </span>
            </div>
          </div>
          <div
            class="frm-in chg"
          >
            <label>
              Filters
            </label>
            <div
              class="action-filters"
            >
              <div>
                <div
                  class="sc-gicCDI lgPhGu frm-in chk"
                  data-for="main-tooltip"
                  data-tip="Newspaper content without audio/video"
                >
                  <div>
                    <input
                      checked=""
                      class="sc-ezWOiH dyWkoI"
                      id="isPrintContent-0"
                      name="isPrintContent"
                      role="none"
                      type="checkbox"
                      value="0"
                    />
                    <label
                      for="isPrintContent-0"
                    >
                      Print Content
                    </label>
                  </div>
                </div>
                <div
                  class="sc-gicCDI lgPhGu frm-in chk"
                  data-for="main-tooltip"
                >
                  <div>
                    <input
                      checked=""
                      class="sc-ezWOiH dyWkoI"
                      id="included-Included"
                      name="included"
                      role="none"
                      type="checkbox"
                      value="Included"
                    />
                    <label
                      for="included-Included"
                    >
                      Included
                    </label>
                  </div>
                </div>
                <div
                  class="sc-gicCDI lgPhGu frm-in chk"
                  data-for="main-tooltip"
                >
                  <div>
                    <input
                      checked=""
                      class="sc-ezWOiH dyWkoI"
                      id="ticker-On Ticker"
                      name="ticker"
                      role="none"
                      type="checkbox"
                      value="On Ticker"
                    />
                    <label
                      for="ticker-On Ticker"
                    >
                      On Ticker
                    </label>
                  </div>
                </div>
              </div>
              <div>
                <div
                  class="sc-gicCDI lgPhGu frm-in chk"
                  data-for="main-tooltip"
                >
                  <div>
                    <input
                      checked=""
                      class="sc-ezWOiH dyWkoI"
                      id="commentary-Commentary"
                      name="commentary"
                      role="none"
                      type="checkbox"
                      value="Commentary"
                    />
                    <label
                      for="commentary-Commentary"
                    >
                      Commentary
                    </label>
                  </div>
                </div>
                <div
                  class="sc-gicCDI lgPhGu frm-in chk"
                  data-for="main-tooltip"
                >
                  <div>
                    <input
                      checked=""
                      class="sc-ezWOiH dyWkoI"
                      id="topStory-Top Story"
                      name="topStory"
                      role="none"
                      type="checkbox"
                      value="Top Story"
                    />
                    <label
                      for="topStory-Top Story"
                    >
                      Top Story
                    </label>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="box"
        >
          <h2
            class="caps"
          >
            Advanced Search
          </h2>
          <div
            style="display: flex; flex-direction: row; min-width: 550px;"
          >
            <div
              class="sc-iBkjds bEngFb sc-himrzO VoNvn frm-in"
              direction="column"
            >
              <label
                class=""
                for="sel-fieldType"
              >
                Field Type
              </label>
              <div
                class="sc-ftvSup klGvED"
                direction="row"
              >
                <div
                  class="sc-gXmSlM dLJUdP css-b62m3t-container"
                  id="sel-fieldType"
                >
                  <span
                    class="css-1f43avz-a11yText-A11yText"
                    id="react-select-4-live-region"
                  />
                  <span
                    aria-atomic="false"
                    aria-live="polite"
                    aria-relevant="additions text"
                    class="css-1f43avz-a11yText-A11yText"
                  />
                  <div
                    class="rs__control css-1s2u09g-control"
                  >
                    <div
                      class="rs__value-container css-319lph-ValueContainer"
                    >
                      <div
                        class="rs__placeholder css-14el2xx-placeholder"
                        id="react-select-4-placeholder"
                      >
                        Select...
                      </div>
                      <div
                        class="rs__input-container css-6j8wv5-Input"
                        data-value=""
                      >
                        <input
                          aria-autocomplete="list"
                          aria-describedby="react-select-4-placeholder"
                          aria-expanded="false"
                          aria-haspopup="true"
                          autocapitalize="none"
                          autocomplete="off"
                          autocorrect="off"
                          class="rs__input"
                          id="react-select-4-input"
                          role="combobox"
                          spellcheck="false"
                          style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                          tabindex="0"
                          type="text"
                          value=""
                        />
                      </div>
                    </div>
                    <div
                      class="rs__indicators css-1hb7zxy-IndicatorsContainer"
                    >
                      <span
                        class="rs__indicator-separator css-1okebmr-indicatorSeparator"
                      />
                      <div
                        aria-hidden="true"
                        class="rs__indicator rs__dropdown-indicator css-tlfecz-indicatorContainer"
                      >
                        <svg
                          aria-hidden="true"
                          class="css-tj5bde-Svg"
                          focusable="false"
                          height="20"
                          viewBox="0 0 20 20"
                          width="20"
                        >
                          <path
                            d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                          />
                        </svg>
                      </div>
                    </div>
                  </div>
                  <input
                    name="fieldType"
                    type="hidden"
                    value=""
                  />
                </div>
              </div>
              <input
                autocomplete="off"
                style="opacity: 0; width: 100%; height: 0px; position: absolute;"
                tabindex="-1"
                value=""
              />
            </div>
            <div
              class="sc-iBkjds bEngFb sc-cCsOjp cTeXVq frm-in"
              direction="column"
            >
              <label
                class=""
                for="txt-searchTerm"
              >
                Search Terms
              </label>
              <div
                class="sc-ftvSup fVbDcX"
                direction="row"
              >
                <input
                  class="sc-ciZhAO fqGkKN txt test"
                  data-for="main-tooltip"
                  name="searchTerm"
                  role="none"
                  type="text"
                  value=""
                  width="100%"
                />
              </div>
            </div>
          </div>
          <div
            class="sc-iBkjds kGkZUV frm-in dateRange"
            direction="column"
          >
            <label
              data-for="main-tooltip"
              data-tip="Date created"
            >
              Date Range
            </label>
            <div
              class="sc-ftvSup klGvED"
              direction="row"
            >
              <div
                class="sc-bZkfAO ddaXFf frm-in"
                width="10em"
              >
                <div
                  class="react-datepicker-wrapper"
                >
                  <div
                    class="react-datepicker__input-container"
                  >
                    <input
                      class="dpk"
                      name="startDate"
                      placeholder="YYYY MM DD"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
              <div
                class="sc-bZkfAO ddaXFf frm-in"
                width="10em"
              >
                <div
                  class="react-datepicker-wrapper"
                >
                  <div
                    class="react-datepicker__input-container"
                  >
                    <input
                      class="dpk"
                      name="endDate"
                      placeholder="YYYY MM DD"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
          <button
            class="sc-jSMfEi cIvmHR btn "
            data-for="main-tooltip"
            name="search"
            type="button"
          >
            <div>
              Search
            </div>
          </button>
          <button
            class="sc-jSMfEi jYpwvk btn "
            data-for="main-tooltip"
            name="clear"
            type="button"
          >
            <div>
              Clear
            </div>
          </button>
        </div>
      </div>
      <div
        class="content-list"
      >
        <div
          class="sc-jqUVSM kyNNLv table"
          role="table"
          style="min-width: 0px;"
        >
          <div
            role="rowheader"
          >
            <div
              class="rh"
              role="row"
              style="display: flex; flex: 1 0 auto; min-width: 0px;"
            >
              <div
                colspan="1"
                role="columnheader"
                style="box-sizing: border-box; flex: 300 0 auto; min-width: 0px; width: 300px; cursor: pointer;"
                title="Toggle SortBy"
              >
                Headline
              </div>
              <div
                colspan="1"
                role="columnheader"
                style="box-sizing: border-box; flex: 150 0 auto; min-width: 0px; width: 150px; cursor: pointer;"
                title="Toggle SortBy"
              >
                Source
              </div>
              <div
                colspan="1"
                role="columnheader"
                style="box-sizing: border-box; flex: 125 0 auto; min-width: 0px; width: 125px; cursor: pointer;"
                title="Toggle SortBy"
              >
                Type
              </div>
              <div
                colspan="1"
                role="columnheader"
                style="box-sizing: border-box; flex: 125 0 auto; min-width: 0px; width: 125px; cursor: pointer;"
                title="Toggle SortBy"
              >
                Section/Page
              </div>
              <div
                colspan="1"
                role="columnheader"
                style="box-sizing: border-box; flex: 125 0 auto; min-width: 0px; width: 125px; cursor: pointer;"
                title="Toggle SortBy"
              >
                Username
              </div>
              <div
                colspan="1"
                role="columnheader"
                style="box-sizing: border-box; flex: 100 0 auto; min-width: 0px; width: 100px; cursor: pointer;"
                title="Toggle SortBy"
              >
                Status
              </div>
              <div
                colspan="1"
                role="columnheader"
                style="box-sizing: border-box; flex: 100 0 auto; min-width: 0px; width: 100px; cursor: pointer;"
                title="Toggle SortBy"
              >
                Date
              </div>
              <div
                colspan="1"
                role="columnheader"
                style="box-sizing: border-box; flex: 50 0 auto; min-width: 0px; width: 50px;"
              >
                Use
              </div>
            </div>
          </div>
          <div
            role="rowgroup"
          >
            <div
              role="row"
              style="display: flex; flex: 1 0 auto; min-width: 0px;"
            >
              <div
                role="cell"
                style="box-sizing: border-box; flex: 300 0 auto; min-width: 0px; width: 300px;"
              >
                <div
                  class="sc-gKXOVf llwlHb ellipsis"
                >
                  test headline
                </div>
              </div>
              <div
                role="cell"
                style="box-sizing: border-box; flex: 150 0 auto; min-width: 0px; width: 150px;"
              >
                <div
                  class="sc-gKXOVf llwlHb ellipsis"
                >
                  TEST
                </div>
              </div>
              <div
                role="cell"
                style="box-sizing: border-box; flex: 125 0 auto; min-width: 0px; width: 125px;"
              >
                <div
                  class="sc-gKXOVf llwlHb ellipsis"
                >
                  Television
                </div>
              </div>
              <div
                role="cell"
                style="box-sizing: border-box; flex: 125 0 auto; min-width: 0px; width: 125px;"
              >
                <div
                  class="sc-gKXOVf llwlHb ellipsis"
                >
                  A32
                </div>
              </div>
              <div
                role="cell"
                style="box-sizing: border-box; flex: 125 0 auto; min-width: 0px; width: 125px;"
              >
                <div
                  class="sc-gKXOVf llwlHb ellipsis"
                >
                  user
                </div>
              </div>
              <div
                role="cell"
                style="box-sizing: border-box; flex: 100 0 auto; min-width: 0px; width: 100px;"
              >
                Publish
              </div>
              <div
                role="cell"
                style="box-sizing: border-box; flex: 100 0 auto; min-width: 0px; width: 100px;"
              >
                05/12/2022
              </div>
              <div
                role="cell"
                style="box-sizing: border-box; flex: 50 0 auto; min-width: 0px; width: 50px;"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 448 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M400 480H48c-26.51 0-48-21.49-48-48V80c0-26.51 21.49-48 48-48h352c26.51 0 48 21.49 48 48v352c0 26.51-21.49 48-48 48zm-204.686-98.059l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.248-16.379-6.249-22.628 0L184 302.745l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.25 16.379 6.25 22.628.001z"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div
            class="sc-kDDrLX gVJsSB"
          >
            <button
              class="sc-jSMfEi czvrZz btn "
              data-for="main-tooltip"
              disabled=""
              name="firstPage"
              type="button"
            >
              <div>
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="m16.293 17.707 1.414-1.414L13.414 12l4.293-4.293-1.414-1.414L10.586 12zM7 6h2v12H7z"
                  />
                </svg>
              </div>
            </button>
            <button
              class="sc-jSMfEi czvrZz btn "
              data-for="main-tooltip"
              disabled=""
              type="button"
            >
              <div>
                1
              </div>
            </button>
            <button
              class="sc-jSMfEi czvrZz btn "
              data-for="main-tooltip"
              disabled=""
              name="lastPage"
              type="button"
            >
              <div>
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M7.707 17.707 13.414 12 7.707 6.293 6.293 7.707 10.586 12l-4.293 4.293zM15 6h2v12h-2z"
                  />
                </svg>
              </div>
            </button>
          </div>
        </div>
      </div>
      <div
        class="content-actions"
      >
        <button
          class="sc-jSMfEi cIvmHR btn "
          data-for="main-tooltip"
          name="create"
          type="button"
        >
          <div>
            Create Snippet
          </div>
        </button>
        <div
          class="addition-actions"
          style="margin-top: 2%;"
        >
          <button
            class="sc-jSMfEi jYpwvk btn "
            data-for="main-tooltip"
            data-tip="Under Construction"
            disabled=""
            name="create"
            type="button"
          >
            <div>
              Send Lois Front Pages
            </div>
          </button>
          <button
            class="sc-jSMfEi jYpwvk btn "
            data-for="main-tooltip"
            data-tip="Under Construction"
            disabled=""
            name="create"
            type="button"
          >
            <div>
              Send Top Stories
            </div>
          </button>
          <button
            class="sc-jSMfEi jYpwvk btn "
            data-for="main-tooltip"
            data-tip="Under Construction"
            disabled=""
            name="create"
            type="button"
          >
            <div>
              Send Send Lois to Commentary
            </div>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
